RecoEgamma/EgammaTools/plugins/EG9X105XObjectUpdateModifier.cc

```diff
@@ -122,8 +131,12 @@ void EG9X105XObjectUpdateModifier::modifyObject(reco::GsfElectron& ele)const
     throw cms::Exception("LogicError") <<" in EG9X105ObjectUpdateModifier, line "<<__LINE__<<" electron "<<ele.et()<<" "<<ele.eta()<<" "<<ele.superCluster()->seed()->seed().rawId()<<" failed to match to the electrons the key map was keyed to, check the map collection is correct";
   }
   reco::GsfElectron::ConversionRejection convRejVars = ele.conversionRejectionVariables();
-  //its rather important to use the core function here to get the org trk ref
-  convRejVars.vtxFitProb = ConversionTools::getVtxFitProb(ConversionTools::matchedConversion(ele.core()->ctfTrack(),*conversions_.handle(),beamspot_.handle()->position(),2.0,1e-6,0));
+  if(allowGsfTrkMatchForConvs_){
+    convRejVars.vtxFitProb = ConversionTools::getVtxFitProb(ConversionTools::matchedConversion(ele.core(),*conversions_.handle(),beamspot_.handle()->position()));
```

Comment:
So @slava77 , very rarely I get different results for 
`ConversionTools::matchedConversion(ele.core(),*conversions_.handle(),beamspot_.handle()->position())` and 
`ConversionTools::matchedConversion(ele.core(),*conversions_.handle(),beamspot_.handle()->position(), 2.0,1e-6,0)); ` where 2.0,1e-6,0 are the default parameters of that function. I tried passing them in as 2.0f, 1e-6f, 0 (incase the 1e-6 had some silly floating point rounding) but it didnt change anything.

Just curious why?

Reply:
The method is 
```c++
    static reco::Conversion const*         matchedConversion(const reco::GsfElectronCore &eleCore,
                                                  const reco::ConversionCollection &convCol, const math::XYZPoint &beamspot, bool allowCkfMatch=true, float lxyMin=2.0, float probMin=1e-6, unsigned int nHitsBeforeVtxMax=0);
```
Do you really mean passing ```2.0``` for the ```allowCkfMatch```?

Reply:
urgh. thats what  I was missing. Fooled by the very similar function below which doesnt have that bool (and has different default parameters so I have to set them to be the same)

Thanks